prepare_iso/create_firstboot_pkg.sh

```diff
@@ -0,0 +1,50 @@
+#!/bin/bash
+
+SCRIPT_DIR="$(cd "$(dirname "$0")"; pwd)"
+SUPPORT_DIR="$SCRIPT_DIR/support"
+
+# Parse the optional command line switches
+USER="vagrant"
+PASSWORD="vagrant"
+IMAGE_PATH="$SUPPORT_DIR/vagrant.jpg"
+
+# Flags
+DISABLE_REMOTE_MANAGEMENT=0
+DISABLE_SCREEN_SHARING=0
+DISABLE_SIP=0
+
+create_firstboot_pkg() {
+  # payload items
+  mkdir -p "$SUPPORT_DIR/pkgroot/private/var/db/dslocal/nodes/Default/users"
+  mkdir -p "$SUPPORT_DIR/pkgroot/private/var/db/shadow/hash"
+  BASE64_IMAGE=$(openssl base64 -in "$IMAGE_PATH")
+  # Replace USER and BASE64_IMAGE in the user.plist file with the actual user and image
+  render_template "$SUPPORT_DIR/user.plist" > "$SUPPORT_DIR/pkgroot/private/var/db/dslocal/nodes/Default/users/$USER.plist"
+  USER_GUID=$(/usr/libexec/PlistBuddy -c 'Print :generateduid:0' "$SUPPORT_DIR/user.plist")
+  # Generate a shadowhash from the supplied password
+  "$SUPPORT_DIR/generate_shadowhash" "$PASSWORD" > "$SUPPORT_DIR/pkgroot/private/var/db/shadow/hash/$USER_GUID"
```

Comment:
It seems like this password is not correctly understood in High Sierra. System will boot, Vagrant user exists, but will never log in.